core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -586,24 +656,37 @@ object ConsumerGroupCommand extends Logging {
 
     def deleteGroups(): Map[String, Errors] = {
       val groupsToDelete = opts.options.valuesOf(opts.groupOpt).asScala.toList
-      val result = adminClient.deleteConsumerGroups(groupsToDelete)
-      val successfullyDeleted = result.filter {
+      val deletedGroups = adminClient.deleteConsumerGroups(
+        groupsToDelete.asJava,
+        withTimeoutMs(new DeleteConsumerGroupsOptions)
+      ).deletedGroups().asScala
+
+      val result = deletedGroups.mapValues { f =>
+        Try(f.get()) match {
+          case _: Success[_] => Errors.NONE
+          case Failure(ee: ExecutionException) if ee.getMessage.contains("TimeoutException") => Errors.COORDINATOR_NOT_AVAILABLE
```

Comment:
This is a bit weird. Can you explain the reasoning?

Reply:
I tried to follow the logic of the previous version of this method: https://github.com/apache/kafka/blob/9ea81baf34113ccb05244ea9fb63199e7f071bb0/core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala#L590-L599

KafkaAdminClient now returns futures as part of DeleteConsumerGroupsResult when you try to delete with deleteConsumerGroups(). Please let me know if you have an idea how to simplify it.

Reply:
I think the part that is unclear to me is why we are checking for TimeoutException in the exception message. It seems like a brittle solution. Can we make it simple and just treat all exceptions as failures?